rasa/nlu/classifiers/regex_message_handler.py

```diff
@@ -0,0 +1,283 @@
+        does not contain real text but the regex we expect here.
+
+        Args:
+            messages: list of messages
+            domain: the domain
+        Returns:
+            list of messages where the i-th message is equal to the i-th input message
+            if that message does not need to be unpacked, and a new message with the
+            extracted attributes otherwise
+        """
+        return [self._unpack(message, domain) for message in messages]
+
+    def _unpack(self, message: Message, domain: Domain) -> Message:
+        """Unpacks the messsage if `TEXT` contains an encoding of attributes.
+
+        Args:
+            message: some message
+            domain: the domain
+        Returns:
+            the given message if that message does not need to be unpacked, and a new
+            message with the extracted attributes otherwise
+        """
+        user_text = message.get(TEXT).strip()
+
+        # If the prefix doesn't match, we don't even need to try to match the pattern.
+        if not user_text.startswith(self._prefix):
+            return message
+
+        # Try to match the pattern.
+        match = self._pattern.match(user_text)
+
+        # If it doesn't match, then (potentially) something went wrong, because the
+        # message text did start with the special prefix -- however, a user might
+        # just have decided to start their text this way.
+        if not match:
+            logger.warning(f"Failed to parse intent end entities from '{user_text}'.")
+            return message
+
+        # Extract attributes from the match - and validate it via the domain.
+        intent_name = self._parse_intent_name(match, domain)
+        confidence = self._parse_optional_confidences(match)
+        entities = self._parse_optional_entities(match, domain)
+
+        # The intent name is *not* optional, but during parsing we might find out
+        # that the given intent is unknown (and warn). In this case, stop here.
+        if intent_name is None:
+            return message
+
+        if match.group("rest"):
+            rasa.shared.utils.io.raise_warning(
+                f"Failed to parse arguments in line '{match.string}'. "
+                f"Failed to interpret some parts. "
+                f"Continuing without {match.group('rest')}. ",
+                docs=DOCS_URL_STORIES,
+            )
+
+        # Add the results to the message.
+        intent_data = {
+            INTENT_NAME_KEY: intent_name,
+            PREDICTED_CONFIDENCE_KEY: confidence,
+        }
+        intent_ranking = [
+            {INTENT_NAME_KEY: intent_name, PREDICTED_CONFIDENCE_KEY: confidence,}
+        ]
+        message_data = {}
+        message_data[INTENT] = intent_data
+        message_data[INTENT_RANKING_KEY] = intent_ranking
+        message_data[ENTITIES] = entities
+        return Message(message_data, output_properties=set(message_data.keys()))
```

Comment:
.... which tokens?

Reply:
the ones from the `Tokenizer`. Should be technically just one 1 token but would be still great to not omit it

Reply:
Where did that happen in the old version? The regex interpreter didn't return tokens

Reply:
@ancalita Did we skip that on purpose back then or did we just miss it?

Reply:
If there ever where tokens, then it's a bit strange that the indices where chosen like [this](https://github.com/RasaHQ/rasa/blob/7625fcaec9a5d1f47dbcc093a405e9369e7261fc/rasa/shared/nlu/interpreter.py#L120) 🤔  ... which is removed atm as well 😬  (because policies etc won't use those anyway - but I'll add them back in with the text now because RasaX)

Reply:
The `RegexInterpreter` definitely never had tokens. This is a very recent changes. If we don't add tokens everytime we need to communicate that to Christof and his team

Reply:
Hmm, good question on why `RegexInterpreter` doesn't add tokens, honestly it didn't occur to me during implementation 🙈 
I was following this [guide ](https://www.notion.so/rasa/Sending-tokens-with-events-to-Rasa-X-05fe8895e4aa416e8ca8e3e1f339bc91)which @joejuzl prepared, would you know why tokens are only added by the `nlu.model.Interpreter`?

Reply:
It can't the messages it gets look like "/<intentname>@<confidencevalue><entity-dict>" - there is no text :)

Due to the worries regarding RasaX, I've just added back in:
- "text" attribute - which contains that malformed message and will need to be caught by TED later then (will open an issue if we merge this eventually :D)
- "start/end" positions for entities - which are (just as before) the start/end of the match of the entity regex

Reply:
@ancalita it was not intentional to not add tokens in the `RegexInterpreter` I just didn't think of it in the scope/implementation proposal. Does it even make sense though? What would the tokens even be? Just a single one fo the intent?

Reply:
> What would the tokens even be? Just a single one fo the intent?

I thought just whatever the tokenizer gives us for this 🤷🏻 We can also omit - in that case Rasa X should fallback to just use the entire text as token? Do we have an issue for Rasa X where we can put that information?

Reply:
I'll add this to [this information](https://github.com/RasaHQ/rasa-x/issues/4704)